### ISSUE TITLE
Continuously running Herbie data ingestion workflow

### ISSUE THREAD

**Issue by razatsutradhar** [microsoft/farmvibes-ai#79]
Hello,
I have been attempting to run the Deep Micro Climate notebook in FarmVibes (https://github.com/microsoft/farmvibes-ai/tree/main/notebooks/deepmc). We have set up a VM instance in azure by following the VM setup instructions in the documentation. When we attempted to run the notebook, it created 4 VibeWorkFlowRuns for data ingestion from Herbie forecast (temperature, humidity, u, v). I let this run for a long time (almost 24 hours) and it keeps saying the workflow is running. I have also tried, restarting the RestAPI with the farmvibes-ai.sh file and also reinstalled everything with the farmvibes-ai.sh file. However, despite this, it still is not finishing the run.  I suspect it might be a VM configuration issue with it's networking or it might be an issue with the RestAPI setup on this VM. Any help is appreciated. Thanks!

**Comment by v-ngangarapu** [comment]
Thanks for reaching out on issue identified.

1. Can you please provide details of time range, you're trying to download.
2. The client pulls information from cache, if request already submitted. Doing restart will not help here. You may need to clear cache also and then start running.
3.  I also suggest, to download data for single day and see how the response is. The day choosen must be outside the time range of the request you submitted previously else it will show same status as running or pending. For a single day, the execution time should be less than a 5 minutes. We observed previously, downloading data for 1 year taking approximately 12 to 18 hours with more than 5 workflow instances
4. If you can share logs for the workflow, it will be helpful.

Thanks

**Comment by razatsutradhar** [comment]
Hey, so I attempted a few things now. I tried downloading it from October 1, 2020 - October 1, 2022. I restarted the project in a new VM and still had the same issues. I have attached the logs from the most recent attempts. "terravibes-worker-9fc897475-h7ngb" is the log for the one-day download set outside of the time range of the other workflow "terravibes-worker-9fc897475-dl8v8" that is for 2 years. 


[terravibes-rest-api.log](https://github.com/microsoft/farmvibes-ai/files/11191597/terravibes-rest-api.log)
[terravibes-cache.log](https://github.com/microsoft/farmvibes-ai/files/11191598/terravibes-cache.log)
[terravibes-worker-9fc897475-h7ngb.log](https://github.com/microsoft/farmvibes-ai/files/11191599/terravibes-worker-9fc897475-h7ngb.log)
[terravibes-worker-9fc897475-dl8v8.log](https://github.com/microsoft/farmvibes-ai/files/11191600/terravibes-worker-9fc897475-dl8v8.log)
[terravibes-orchestrator.log](https://github.com/microsoft/farmvibes-ai/files/11191601/terravibes-orchestrator.log)

**Comment by brsilvarec** [comment]
Hey @razatsutradhar,

    Thanks for providing the logs. Can you please upgrade FarmVibes.AI to the latest version? We just released a new version that periodically checks the cluster health and restart broken services. If the problem persists, please let us know.

Best Regards,

**Comment by razatsutradhar** [comment]
Hey, 
Thanks for the response. This has solved some similar issues I had in the heatmaps notebook. I did notice that for the heatmaps notebook, if a heatmap is generated with the parameter ```"data_scale": False``` instead of True, it would lead to an assertion error later on: 
```File "/app/ops/heatmap_sensor/soil_sample_heatmap.py", line 69, in create_heatmap
   assert scaler is not None, "Scaler is None"
    AssertionError: Scaler is None
```
I just set the set data_scale to true to make it work. 
Currently running the deepmc notebook again and I will let you know if it works. 
Thanks!
